third_party/protobuf/protobuf-3.4.0.patch

```diff
@@ -0,0 +1,295 @@
+diff -rup protobuf-3.4.0.orig/configure.ac protobuf-3.4.0/configure.ac
```

Comment:
what is this patch file for?

Reply:
this patch is to ensure we use a fast path for C++11x so that performance is not affected. The patch was originally done by @congwang - just redid it for protobuf 3.4.0